iot-hub/Samples/device/DeviceReconnectionSample/Program.cs

```diff
@@ -22,11 +22,15 @@ public class Program
         // - create a launchSettings.json (see launchSettings.json.template) containing the variable
         private static string s_deviceConnectionString = Environment.GetEnvironmentVariable("IOTHUB_DEVICE_CONN_STRING");
 
-        // Select one of the following transports used by DeviceClient to connect to IoT Hub.
-        private static TransportType s_transportType = TransportType.Amqp;
-        //private static TransportType s_transportType = TransportType.Mqtt;
-        //private static TransportType s_transportType = TransportType.Amqp_WebSocket_Only;
-        //private static TransportType s_transportType = TransportType.Mqtt_WebSocket_Only;
+        // Specify one of the following transports used by DeviceClient to connect to IoT Hub.
+        //   Amqp
+        //   Http1
+        //   Amqp_WebSocket_Only
+        //   Amqp_Tcp_only
+        //   Mqtt
+        //   Mqtt_WebSocket_Only
+        //   Mqtt_Tcp_Only
+        private static readonly string s_transportType = Environment.GetEnvironmentVariable("IOTHUB_DEVICE_TRANSPORT_TYPE");
 
         public static async Task<int> Main(string[] args)
         {
```

Comment:
@abhipsaMisra on line 47 we instantiate a ConsoleEventListener, but never use it. I didn't want to remove it until I asked you what the point was.

Reply:
For the console event listener, we only need to initialize an instance of that class. Since we define custom implementations of the write events from `EventListener`, the appropriate action is taken whenever our library code writes to event source. I cannot say that I am super confident in how EventSource works, but this has been my finding.